app/components/Form.js

```diff
@@ -0,0 +1,237 @@
+import React from 'react';
+import { StyleSheet, Text, View, TextInput, TouchableOpacity, AsyncStorage} from 'react-native';
+import { StackNavigator } from 'react-navigation';
+
+export default class Form extends React.Component {
+
+    constructor(props){
+        super(props);
+        this.state={
```

Comment:
Its better if you handle this state in the screen and only pass the data as props.